controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1433,6 +1453,20 @@ func (r *BareMetalHostReconciler) doServiceIfNeeded(prov provisioner.Provisioner
 		dirty = true
 	}
 
+	if hfcDirty && started {
+		hfcDirty, err = r.saveHostFirmwareComponents(prov, info, hfc)
+		if err != nil {
+			return actionError{errors.Wrap(err, "could not save the host firmware components")}
+		}
+
+		if hfcDirty {
+			if err := r.Status().Update(info.ctx, hfc); err != nil {
+				return actionError{errors.Wrap(err, "failed to update hostfirmwarecomponents status")}
+			}
+		}
+		dirty = true
```

Comment:
The BMH is no longer dirty: you've just saved its status (see above). Maybe you actually only need this `dirty = true` but do not need the Update above?

Reply:
agree that the BMH is not dirty, but unsure if I don't need the Update above

Reply:
Done